app/models/fae/navigation.rb

```diff
@@ -69,9 +72,28 @@ def item(text, options={})
       # prefer `subitems` over `sublinks` for new DSL but still need to support old structure in v1
       hash[:sublinks] = hash[:subitems] if hash[:subitems].present?
 
+      # authorize link with current user
+      unless can_view_path(hash[:nested_path])
+        hash[:path] = '#'
```

Comment:
The item is removed in the `recursive_authorization` method. This will flag the item for removal but also makes sure `Fae::Navigation#strucrture` never contains a path the `current_user` can't view.